lib/streamlit/connections/base_connection.py

```diff
@@ -24,30 +26,43 @@ class BaseConnection(ABC, Generic[T]):
     """TODO(vdonato): docstrings for this class and all public methods."""
 
     def __init__(self, connection_name: str = "default", **kwargs) -> None:
+        if connection_name == "default":
+            connection_name = self.default_connection_name()
+
         self._connection_name = connection_name
         self._kwargs = kwargs
 
         self._raw_instance: Optional[T] = self.connect(**kwargs)
+        self._config_section_hash = calc_md5(pickle.dumps(self.get_secrets()))
```

Comment:
nit: are you sure that pickle is suitable for hashing in this case? Without having looked closer into it, I assume there could be other aspects that could influence the hash of a pickle despite just differences in the values. A potential alternative could be `json.dumps` here which might make it a bit more deterministic.

Reply:
Ah, yeah -- I did a bit of research and found that there are some cases where pickle's return value may change.

I'm not entirely sure whether these issues will come up in practice over the life of a Streamlit server (I'd actually lean toward thinking that they won't), but there's no downside to using `json.dumps` instead, so we might as well switch to that to be safe.